### PR TITLE
Bacpop-113 failed sample

### PR DESCRIPTION
**Pull request by absternator**
This PR contains work for updating beebop_py to use quality control as per #39. These should both reviewed together merged together. 

The work here contains basic error handling that was need to accommodate qc updates. Once PopPUNK changes are in (writing qc failures to file) we can display these. The intent is to display them as tooltips on hover of failed tag. Current PR defaults to 'Invalid Sample' for the tooltip.

Note: After review, revert API branch back to main (after beebop_py is merged)

Testing:
the [ticket](https://mrc-ide.myjetbrains.com/youtrack/issue/bacpop-113/Sample-quality-feedback) contains failed samples that can be used. Put these in and they should fail (except 2). This allows some to fail and others to pass. 
Feel free to mess around with combinations of failed and passing. or even just failed ones.

### CODE REVIEW

**Review by EmmaLRussell — COMMENTED** [review]
Looks good to me, tiny suggestion below and a couple of suggestions. 

On behaviour, after the cluster assignment has failed, it still shows "started" and "deferred" respectively in the Network and Microreact columns for a while, and then those are eventually replaced by the download/visit buttons, but they're disabled. I think it might be a clearer user experience if the buttons were replaced by "failed" badges when those jobs also fail. 

But also - is there any point in continuing to poll for Network or Microreact when the cluster assignment fails? Isn't cluster assignment a pre-requisite for the other two? Does it even make sense to kick off those jobs when cluster assignment fails?

File: app/client-v2/src/components/ProjectView/ProjectPostRun.vue [hunk 1]
```diff
@@ -50,8 +50,18 @@ const getMicroreactSettingsTooltip = () => {
...
+              <Tag
+                v-else-if="
+                  projectStore.project.status?.assign === 'failed' ||
+                  (projectStore.project.status?.assign === 'finished' && !data.cluster)
+                "
+                v-tooltip.top="`${data.failReason ?? 'Invalid Sample'}`"
+                value="failed"
+                severity="danger"
+              />
+              <Tag v-else value="pending" severity="warning" />
```
[the remark this answers is not in the document]
Comment (absternator) [reply app/client-v2/src/components/ProjectView/ProjectPostRun.vue:60]: nope this is the pending state (waiitng to assign cluster)

File: app/client-v2/src/components/ProjectView/ProjectPostRun.vue [hunk 2]
```diff
@@ -50,8 +50,18 @@ const getMicroreactSettingsTooltip = () => {
           <Column field="cluster" header="Cluster">
             <template #body="{ data }">
               <span v-if="data.cluster"> {{ data.cluster }}</span>
-              <Tag v-else value="pending" severity="warning" /> </template
-          ></Column>
+              <Tag
+                v-else-if="
+                  projectStore.project.status?.assign === 'failed' ||
+                  (projectStore.project.status?.assign === 'finished' && !data.cluster)
```
[the remark this answers is not in the document]
Comment (absternator) [reply app/client-v2/src/components/ProjectView/ProjectPostRun.vue]: as per pr comment below

**Comment by absternator** [comment]
> Looks good to me, tiny suggestion below and a couple of suggestions.
> 
> On behaviour, after the cluster assignment has failed, it still shows "started" and "deferred" respectively in the Network and Microreact columns for a while, and then those are eventually replaced by the download/visit buttons, but they're disabled. I think it might be a clearer user experience if the buttons were replaced by "failed" badges when those jobs also fail.
> 
> But also - is there any point in continuing to poll for Network or Microreact when the cluster assignment fails? Isn't cluster assignment a pre-requisite for the other two? Does it even make sense to kick off those jobs when cluster assignment fails?

1. Agreed and done 
2. The behaviour is there for multiple samples... as some can fail and some not... thus the status isn't failed in that case, but they just don't have clusters assigned to them. (i.e status can be finished (for project) but cluster can be undefined (for sample)..

**Review by EmmaLRussell — APPROVED** [review]
Yeah, this is much clearer thanks! looks like buildkite failed, assume that's just something ephemeral?

**Pull request merged**